prometheus-to-sd/main.go

```diff
@@ -113,7 +121,18 @@ func main() {
 		glog.Error(http.ListenAndServe(fmt.Sprintf(":%d", *debugPort), nil))
 	}()
 
-	client := oauth2.NewClient(context.Background(), google.ComputeTokenSource(""))
+	var client *http.Client
+
+	if *projectOverride != "" {
+		client, err = google.DefaultClient(context.Background(), "https://www.googleapis.com/auth/cloud-platform")
```

Comment:
I'm a little bit unfamiliar with how authentication works here, but how this is supposed to work? Shouldn't we pass some token for correct authorization?

Reply:
This uses Application Default Credentials, taken from the environment variable. To use it, the user needs to make sure GOOGLE_APPLICATION_CREDENTIALS points to the path of a file with credentials. (this might be useful: https://godoc.org/golang.org/x/oauth2/google#DefaultTokenSource)